Flaky test: SqliteEndToEndSpec fails with 'no such table: journal'

## Description

`SqliteEndToEndSpec` intermittently fails on CI with `SQLite Error 1: SQL logic error - no such table: journal`.

## Error Details

```
[ERROR][Thread 0014][FlowShape`2...](akka://SqliteEndToEndSpec/system/StreamSupervisor-470)]
An exception occurred inside SelectAsync while executing Task. Supervision strategy: Stop
Cause: code = Error (1), message = System.Data.SQLite.SQLiteException (0x87AF001F): SQL logic error
no such table: journal
   at System.Data.SQLite.SQLite3.Prepare(...)
   at LinqToDB.Data.DataConnection.ExecuteReaderAsync(...)
   at Akka.Persistence.Sql.Query.Dao.BaseByteReadArrayJournalDao.<>c.<<JournalSequence>b__7_1>d.MoveNext()
```

Additionally, a stream materialization race condition occurs during cleanup:
```
[ERROR] Error in stage [FlattenMerge(1)]: Attempted to call Materialize() after the ActorMaterializer has been shut down.
```

## Analysis

The test uses `auto-initialize = true` (default) which should create tables, but the table doesn't exist when the query runs. This could be:

1. A race condition in table initialization
2. Test isolation issues with the shared in-memory SQLite database
3. Timing-related issues with xUnit fixtures

## Test Location

- `src/Akka.Persistence.Sql.Tests/Sqlite/SqliteEndToEndSpec.cs`
- `src/Akka.Persistence.Sql.Tests/SqlEndToEndSpecBase.cs`

## Related

Similar to #550 (SqlServerDataOptionsEndToEndSpec flakiness)

## Labels

bug

--- a/src/Akka.Persistence.Sql.Tests/SqlEndToEndSpecBase.cs
+++ b/src/Akka.Persistence.Sql.Tests/SqlEndToEndSpecBase.cs
@@ -85,8 +85,16 @@ akka.persistence {
             _persistenceActor = Sys.ActorOf(Props.Create(() => new MyPersistenceActor(PId)), "persistence-actor-1");
         }
 
-        public Task DisposeAsync()
-            => Task.CompletedTask;
+        public async Task DisposeAsync()
+        {
+            // Properly shut down the actor system to ensure all streams, actors, and materializers
+            // are terminated before the SQLite in-memory database connection is closed.
+            // Without this, there's a race condition where:
+            // 1. xUnit disposes SqliteContainer, closing the in-memory database
+            // 2. Actor system continues running, streams try to query
+            // 3. "no such table" or "Materializer has been shut down" errors occur
+            await Sys.Terminate();
+        }
 
         /// <summary>
         /// Initializes a new <see cref="TestOutputLogger"/> used to log messages.

